corva/api.py

```diff
@@ -0,0 +1,114 @@
+import posixpath
+import re
+from typing import Optional
+
+import requests
+
+
+class Api:
+    """Provides a convenient way to access Corva API and Corva Data API
+
+    Api is a thin wrapper around `requests` library that adds
+     authorization, convenient url usage  and timeouts to requests.
+    """
+
+    TIMEOUT = 30  # seconds
+
+    def __init__(
+        self,
+        *,
+        api_url: str,
+        data_api_url: str,
+        api_key: str,
+        app_name: str,
+        timeout: Optional[int] = None,
+    ):
+        self.api_url = api_url
+        self.data_api_url = data_api_url
+        self.api_key = api_key
+        self.app_name = app_name
+        self.timeout = timeout or self.TIMEOUT
+
+    @property
+    def auth_headers(self):
+        return {
+            'Authorization': f'API {self.api_key}',
+            'X-Corva-App': self.app_name,
+        }
+
+    def get(self, path: str, **kwargs):
+        return self._request('GET', path, **kwargs)
+
+    def post(self, path: str, **kwargs):
+        return self._request('POST', path, **kwargs)
+
+    def patch(self, path: str, **kwargs):
+        return self._request('PATCH', path, **kwargs)
+
+    def put(self, path: str, **kwargs):
+        return self._request('PUT', path, **kwargs)
+
+    def delete(self, path: str, **kwargs):
+        return self._request('DELETE', path, **kwargs)
+
+    def _get_url(self, suffix: str):
+        """Builds complete url from base prefix and suffix
+
+        returns:
+         1 suffix param, if suffix is a complete url
+         2 data api url, if suffix contains data api url pattern
+         3 corva api url, if above points are False
+        """
+
+        if suffix.startswith('http'):
+            return suffix
+
+        suffix = suffix.lstrip(
+            '/'
+        )  # delete leading forward slash for posixpath.join to work correctly
+
+        # search text like api/v1 or api/v10 in path
+        if bool(re.search(r'api/v\d+', suffix)):
```

Comment:
Do we have some unit tests to test the cases here? I can provide a list of URLs and which API they should point to, but I am not sure this properly matches.

Reply:
we have tests, take a look https://github.com/corva-ai/python-sdk/blob/8893009db43743df311eccd3416279338b6b8e29/tests/test_api.py#L44